tests/tests_args_parser.c

```diff
@@ -0,0 +1,93 @@
+#define MUNIT_ENABLE_ASSERT_ALIASES
+#include "munit/munit.h"
+#include "args_parser.h"
+#include "vm/lib/tests_helper.h"
+#include "vm/lib/vm_string.h"
+#include "stdio.h"
+
+
+static char* args_parse_params[] = {
+    /* Flag Name! Flag Value(ARGC/2 TIMES)/
+        Input Argv/ Input Argc */
+    (char*) "filename! programa_legal/ filename! programa_legal/ 3/",
+    (char*) "help! / help/ 2",
+    (char*) "version! / version/ 2",
+    (char*) "execute! programa_legal/ execute! programa_legal/ 3",
+    (char*) "inform! programa_legal/ inform! programa_legal/ 3",
+    NULL
+};
+
+static MunitParameterEnum test_args_parse_params[] = {
+    { (char*) "args_parse_params", args_parse_params },
+    { NULL, NULL },
+};
+
+static MunitResult
+test_args_parse(const MunitParameter params[], void* user_data) {
+    const char* args_parse_params;
+    char* delimiter;
+    char** input_argv;
+    args_t expected_result;
+    args_t predicted_result;
+    int parameters_length, input_argc;
+    char** parsed_args, **parsed_args_t;
+    int position;
+    (void) user_data;
+
+    args_parse_params = munit_parameters_get(params, "args_parse_params");
+
+    // Let's first parse the params
+    parameters_length = strlen(args_parse_params);
+    parsed_args = arg_str_to_array_of_str(args_parse_params, '/');
+    input_argv = arg_str_to_array_of_str(parsed_args[1], '!');
+    input_argc = atoi(parsed_args[2]);
+    parsed_args_t = arg_str_to_array_of_str(parsed_args[0], '!');
+    expected_result.flags->flag_name = parsed_args_t[0];
+    expected_result.flags->flag_value = parsed_args_t[1];
+    // Now we have parsed the arguments, let's test our function
+
+    predicted_result = args_parse(
+        input_argc, input_argv);
+
+    assert_string_equal(
+        expected_result.flags->flag_name,
+        predicted_result.flags->flag_name);
+    if(vm_strcmpl(predicted_result.flags->flag_name, ""))
```

Comment:
This `if` statement is quite weird. Why it doesn't have braces and why there's no indentation of the assert just a line below it? Should the assert be inside the scope of the `if` or not?